src/execution_plan/ops/op_update.c

```diff
@@ -181,13 +186,16 @@ static Record UpdateConsume(OpBase *opBase) {
 		 * for later execution. */
 		EntityUpdateEvalCtx *update_expression = op->update_expressions;
 		for(uint i = 0; i < op->update_expressions_count; i++, update_expression++) {
-			SIValue new_value = SI_CloneValue(AR_EXP_Evaluate(update_expression->exp, r));
-			// Make sure we're updating either a node or an edge.
+			// Get the type of the entity to update.
 			RecordEntryType t = Record_GetType(r, update_expression->record_idx);
+			// If the expected entity was not found, make no updates but do not error.
+			if(t == REC_TYPE_UNKNOWN) continue;
+			// Make sure we're updating either a node or an edge.
 			assert(t == REC_TYPE_NODE || t == REC_TYPE_EDGE);
```

Comment:
I feel like this is going to be an open issue at some point, consider emitting a run-time error.